lib/OpenQA/Git/MaintenanceOutage.pm

```diff
@@ -0,0 +1,31 @@
+package OpenQA::Git::MaintenanceOutage;
+use Mojo::Base -strict, -signatures;
+use Mojo::File 'path';
+use Time::Seconds;
+
+my $DEFAULT_STATE_FILE = "/var/lib/openqa/share/gitlab_maintenance.lock";
+my $MAX_DURATION_SECONDS = 1800;
+
+sub _state_file {
+    return $ENV{OPENQA_GITLAB_MAINTENANCE_FILE} || $DEFAULT_STATE_FILE;
+}
+
+sub decide_outcome {
+    my $state_file = _state_file();
+
+    unless (-f $state_file) {
+        path($state_file)->touch;
+        return {skip => 1};
```

Comment:
Maybe it would make more sense if this function returned a single value from a set of previously defined constants.